### DISCUSSION TITLE
Automatic machine translation with OpenAI does not succeed

### DISCUSSION THREAD

**Discussion by JohnRDOrazio** [WeblateOrg/weblate#11922] [Q&A] [answered]
When attempting to autotranslate a component using OpenAI, I'm getting lock errors. The progress bar at the top of the screen doesn't make any progress, and if I inspect the Chrome developer tools I can see the requests being sent, with `null` results and progress `0`, and a message 'Lock on component-update-lock-59 could not be acquired in 5s`. However I believe that there are communications being sent to and answers received from OpenAI because my bill did go up 😄 

Is there anything I can / should do to get this to work?

**Comment by nijel** [comment]
The lock is obtained before fetching from machine translation (and is retried). Maybe there was another machine translation executed at the same time?

**Comment by JohnRDOrazio** [comment]
The first time I attempted, I had set Weblate + OpenAI as machine transalations. I'm thinking that maybe they were interfering with each other? So then I tried Weblate alone, and it progressed (without many results). Then I tried OpenAI alone, again I got the lock.

I waited a day, and tried again today, this time it seems like there wasn't a lock but it took quite a while to start seeing 1% progress on the progressbar (5-10 mins?). I let it run for about an hour, came back to check, and the process was interrupted with a message along the lines of "couldn't parse response". So in the end, 0% progress 😮

**Reply by nijel** [reply, marked as answer]
Since b4d31853d2748dcb58813037163fb56219a57e64 it should not crash in middle, it would just skip the responses it cannot parse.

**Reply by JohnRDOrazio** [reply]
Great! I updated from Weblate 5.5 to Weblate 5.6, and retried the automatic translation. It stalled initially with the lock error, but I see that in the meantime a few hours later that it has made a lot of progress in the background.
